elements/nuxeo-document-bulk-actions/nuxeo-bulk-widget.js

```diff
@@ -0,0 +1,192 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { mixinBehaviors } from '@polymer/polymer/lib/legacy/class.js';
+import '@polymer/paper-item/paper-item.js';
+import { I18nBehavior } from '@nuxeo/nuxeo-ui-elements/nuxeo-i18n-behavior.js';
+import '@nuxeo/nuxeo-ui-elements/widgets/nuxeo-select.js';
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+
+/**
+`nuxeo-bulk-widget`
+@group Nuxeo UI
+@element nuxeo-bulk-widget
+*/
+class NuxeoBulkWidget extends mixinBehaviors([I18nBehavior], Nuxeo.Element) {
+  static get is() {
+    return 'nuxeo-bulk-widget';
+  }
+
+  static get template() {
+    return html`
+      <style>
+        :host {
+          display: flex;
+          flex-direction: column;
+        }
+        .header {
+          display: flex;
+          align-items: center;
+          justify-content: space-between;
+        }
+        .warning,
+        .error {
+          /* ensures that the bulk widget takes the same vertical space regardless of having a message or not */
+          min-height: 20px;
+        }
+        .warning {
+          color: var(--nuxeo-text-default);
+        }
+        .error {
+          color: var(--nuxeo-warn-text);
+        }
+        label {
+          @apply --nuxeo-label;
+        }
+        nuxeo-select {
+          /* disable underline */
+          --paper-input-container-underline: {
+            display: none;
+          }
+          /* disable underline (focused) */
+          --paper-input-container-underline-focus: {
+            display: none;
+          }
+          /* remove padding on the paper-dropdown-menu */
+          --nuxeo-select-dropdown-menu-padding: 0;
+          /* remove padding on the paper-input */
+          --nuxeo-select-input-container-padding: 0;
+          /* tweak text inside the paper-input element */
+          --paper-input-container-input: {
+            font-size: 12px;
+            text-align: right;
+          }
+          /* tweak text inside paper-items */
+          --paper-item: {
+            font-size: 12px;
+          }
+          /* set the width to make sure the text of the options doesn't wrap */
+          width: 180px;
+        }
+        paper-item[disabled] {
+          @apply --nx-button-text-disabled;
+        }
+      </style>
+      <div class="header">
+        <label>[[label]]</label>
+        <nuxeo-select selected="{{updateMode}}" attr-for-selected="id">
+          <paper-item id="keep" label="Don't change value(s)">Don't change value(s)</paper-item>
+          <paper-item id="replace" label="Replace with">Replace with</paper-item>
+          <paper-item id="remove" label="Remove value(s)" disabled$="[[_required]]">Remove value(s)</paper-item>
+        </nuxeo-select>
+      </div>
+      <slot></slot>
+      <label class$="[[_messageClass]]">[[_message]]</label>
```

Comment:
We should consider to wrap these form inputs in a fieldset element, in order for this label to make sense in this context

Reply:
Label requires always a for attribute. In this case, we need to replace this element with a simple `span`
```suggestion
      <span class$="[[_messageClass]]">[[_message]]</span>
```